File: phalcon/app/model/SysMenuAction.php
<?php
namespace app\model;

use app\library\Safety;

class SysMenuAction extends Model{

  protected $id;
  protected $name;
  protected $action;
  protected $perm;

  /* 构造函数 */
  public function initialize(){
    $this->setSource('sys_menus_action'); //数据表
  }

  /* 名称 */
  function setName($val){
    $num = mb_strlen($val,'utf-8');
    if($num<2 || $num>6){
      self::error('名称为2~6位字符!');
    }
    $this->name = $val;
  }
  function getName(){
    return $this->name;
  }

  /* 命名 */
  function setAction($val){
    if(!Safety::test('^[a-zA-Z]{2,16}$',$val)){
      self::error('命名为2~16位英文!');
    }
    $this->action = $val;
  }
  function getAction(){
    return $this->action;
  }

  /* 权限 */
  public function setPerm($val){
    if((int)$val<2 || (int)$val%2!=0){
      self::error('权限为2的n次方!');
    }
    $this->perm = $val;
  }
  public function getPerm(){
    self::error('测试');
    return $this->perm;
  }

}

File: python/app/model/SysMenuAction.py
from app.model.Model import Model
from app.library.Safety import Safety

# 系统菜单表
class SysMenuAction(Model):

  id = None
  name = ''
  action = ''
  perm = '0'
  ico = ''

  # 构造函数
  def __init__(self):
    self.setSource('sys_menus_action')

  # 名称
  def setName(self,val) :
    num = len(val)
    if num<2 or num>6 :
      self.error('名称为2~6位字符!')
    self.name = val
  def getName(self):
    return self.name

  # 命名
  def setAction(self,val) :
    if not Safety.test(r'^[a-zA-Z]{2,16}$',val) :
      self.error('命名为2~16位英文!')
    self.action = val
  def getAction(self):
    return self.action

  # 权限
  def setPerm(self,val) :
    if val=='' or int(val)<2 or int(val)%2!=0 :
      self.error('权限为2的n次方!')
    self.perm = val
  def getPerm(self):
    return self.perm
